File: Data Structures & Algorithms/maximum-subarray/submission-4.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        m = max(nums)
        while nums and nums[0] < 0:
            nums.remove(nums[0])
        if not nums:
            return m
        l = [0]
        for num in nums:
            l.append(l[-1] + num)
        ma = max(l)
        mi = min(l)
        return ma - mi

File: Data Structures & Algorithms/duplicate-integer/submission-4.swift
class Solution {
    func hasDuplicate(_ nums: [Int]) -> Bool {
        var set: Set<Int> = []
        for n in nums {
            if set.contains(n) {
                return true
            } else {
                set.insert(n)
            }
        }
        return false
    }
}
